client/windows/CCastleInterface.cpp

```diff
@@ -842,7 +842,12 @@ void CCastleBuildings::enterDwelling(int level)
 
 void CCastleBuildings::enterToTheQuickRecruitmentWindow()
 {
-	GH.pushIntT<QuickRecruitmentWindow>(town, pos);
+	const auto hasSomeoneToRecruit = std::any_of(town->creatures.cbegin(), town->creatures.cend(),
+		[](const auto & creatureInfo) { return creatureInfo.first > 0; });
+	if(hasSomeoneToRecruit)
+		GH.pushIntT<QuickRecruitmentWindow>(town, pos);
+	else
+		CInfoWindow::showInfoDialog("There are no creatures to recruit", {});

```

Comment:
Harcdcoding any GUI text is questionable, can you use some config file?

`There's already similar game message displayed on visiting empty dwellings in the advmap, but it requires creature ID. Would it be possible to reuse somehow? https://github.com/vcmi/vcmi/blob/develop/lib/mapObjects/CGTownInstance.cpp#L365`

Nah, this one doesn't fit.

Reply:
thanks, I'll check how to handle that